### ISSUE TITLE
Hosted Providers

### ISSUE THREAD

**Issue by ankane** [pgvector/pgvector#54]
Add hosted providers that support pgvector to this thread.

Initial list:

- [Supabase](https://supabase.com/docs/guides/database/extensions)

**Comment by gregnr** [comment]
@ankane Funny, I was planning to send a PR shortly to request adding Supabase to the list. Been working with their team to get that integrated:
https://github.com/supabase/postgres/pull/472

Thanks again for making a great extension. Currently using this to store and query OpenAI Embeddings.

**Comment by ankane** [comment]
Awesome, big thanks for adding it!

**Comment by m-abdelwahab** [comment]
@ankane. [Neon](neon.tech) recently added support for pgvector: https://neon.tech/docs/release-notes/2023-02-14

**Comment by ankane** [comment]
Thanks @m-abdelwahab, great to hear!

**Comment by Winslett** [comment]
[Crunchy Bridge](https://docs.crunchybridge.com/extensions-and-languages/extensions/)

**Comment by george-bitdotio** [comment]
Howdy! [bit.io](https://bit.io) now [supports pgvector](https://docs.bit.io/docs/supported-sql) :)

**Comment by seanaguinaga** [comment]
Need that Google CloudSQL 8)

**Comment by Vonng** [comment]
[Pigsty](https://github.com/Vonng/pigsty/blob/master/docs/FEATURE.md) - Free RDS PG Alternative

now add pgvector as default extension in [v2.0.2](https://github.com/Vonng/pigsty/releases/tag/v2.0.2)

**Comment by bbedward** [comment]
We added the extension to spilo to use with postgres-operator in kubernetes.

https://github.com/stablecog/spilo

Can just use the docker image in the helm chart values, or build your own:

https://hub.docker.com/r/stablecog/spilo

15.3+ has pgvector for Postgres 11-15

**Comment by pthieu** [comment]
waiting on that RDS

**Comment by gasabr** [comment]
yandex cloud provides an extension too https://cloud.yandex.com/en/docs/managed-postgresql/operations/extensions/cluster-extensions

**Comment by mmmaia** [comment]
> anyone has setup `pg_vector` on Railway? Didn't found anything about installing extensions there :/

@NicolasLopes7  It's not possible yet. You can vote and comment on these feature requests:

- https://feedback.railway.app/feature-requests/p/install-postgres-extensions
- https://feedback.railway.app/feature-requests/p/pgvector-support

**Comment by mwmeyer** [comment]
No support for digital ocean yet. Heres an upvote support request for others that want it: https://ideas.digitalocean.com/app-framework-services/p/saving-openai-embeddings-with-pgvector

**Comment by mfreed** [comment]
[Timescale](https://www.timescale.com/) supports pgVector in its cloud offering.

<img width="700" alt="image" src="https://user-images.githubusercontent.com/205086/235531541-eeb427a5-e075-44bc-9c2f-a18ffa138e08.png">

**Comment by arjun810** [comment]
Available for RDS now! https://aws.amazon.com/about-aws/whats-new/2023/05/amazon-rds-postgresql-pgvector-ml-model-integration/

**Comment by montanalow** [comment]
Also available on hosted [PostgresML](https://postgresml.org/) databases.

**Comment by mulander** [comment]
pgvector is now available on Azure:

* Azure Database for PostgreSQL Flexible Server

Follow [this page](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-extensions) to enable `vector` in the allowed extensions and then simply `CREATE EXTENSION vector;` on your Flexible Server.

* Azure Cosmos DB for PostgreSQL

Just connect to your cluster and issue `SELECT create_extension('vector');`

Main readme update submitted as https://github.com/pgvector/pgvector/pull/129

**Comment by jcoc611-gvt** [comment]
@mulander Seems like extension `vector` is not yet available in Azure Flexible Server (also not present in [this doc](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-extensions#postgres-14-extensions)). Do you know when it will be in general availability?

**Comment by mulander** [comment]
> @mulander Seems like extension `vector` is not yet available in Azure Flexible Server (also not present in [this doc](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-extensions#postgres-14-extensions)). Do you know when it will be in general availability?

@jcoc611-gvt  What region are you checking and is it on a new or deployed cluster? The documentation update (including the doc you linked) is scheduled to be published on the 23rd of May.

**Comment by atomkirk** [comment]
Please upvote the issue to request this extension on GCP Cloud SQL https://issuetracker.google.com/issues/278521774

**Comment by jcoc611-gvt** [comment]
> > @mulander Seems like extension `vector` is not yet available in Azure Flexible Server (also not present in [this doc](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-extensions#postgres-14-extensions)). Do you know when it will be in general availability?
> 
> @jcoc611-gvt What region are you checking and is it on a new or deployed cluster? The documentation update (including the doc you linked) is scheduled to be published on the 23rd of May.

@mulander Seems like new clusters can enable `vector` whereas existing ones cannot.

In any case, it seems like pgvector is crashing on INSERT or UPDATE large vectors (e.g. 1536 dimensions). Logs look something like this:
```
LOG: server process (PID 3517) was terminated by signal 4: Illegal instruction
```

Even on a simple table such as
```
CREATE TABLE test (id bigserial PRIMARY KEY,  v vector(1536));
```

This works fine for trivially small vectors e.g. 1-2 dimensions.

Perhaps pgvector was compiled with optimizations not supported in Azure flexible database environment? Or is it a matter of configuring server parameters/using a SKU with more RAM?

**Comment by ankane** [comment]
@jcoc611-gvt I think it'd be good to move this to an Azure support channel. `Illegal instruction` typically means pgvector was compiled with AVX instructions that aren't available on the machine (and will only appear with vectors of 8+ or 16+ dimensions).

**Comment by aytekinar** [comment]
Hi @jcoc611-gvt and @ankane,

I am from the Azure Cosmos DB for PostgreSQL team, and I am also in touch with @mulander and Azure Database for PostgreSQL Flexible Server devs in our internal thread, investigating the issue.

We also believe the offending line is [`OPTFLAGS = -march=native`](https://github.com/pgvector/pgvector/blob/master/Makefile#L12), and we will try overriding it to avoid any potential issues in the future, especially when/if the extension is built on one machine/architecture and distributed to other machines (this happens when the span of supported VM sizes is large).

For us to investigate the issue and help you better and faster, @jcoc611-gvt, could you please share with us which VM size you were using when you tested the below excerpt:

```pgsql
CREATE TABLE test (id bigserial PRIMARY KEY,  v vector(1536));
```

**Comment by ankane** [comment]
Hi @aytekinar, please create a new issue for this.

**Comment by louis030195** [comment]
https://github.com/different-ai/embedbase hosted version uses pgvector in supabase :)

**Comment by mareksuscak** [comment]
It seems to me that Google's has recently secretly added support for `pgvector 0.4.3` in `POSTGRES_15_2.R20230530.01_00` maintenance update! The [list of extensions in the documentation](https://cloud.google.com/sql/docs/postgres/extensions#miscellaneous-extensions) has not been updated yet to reflect this but running `SELECT * FROM pg_available_extensions;` on a recently updated Cloud SQL instance shows this:

![CleanShot 2023-06-19 at 16 53 53](https://github.com/pgvector/pgvector/assets/613647/ced23475-483c-4032-b5b7-b5984b85f0ec)

Yas!

**Comment by packi** [comment]
We've added pgvector to our supported extensions at Aiven, starting from PG13, see https://docs.aiven.io/docs/products/postgresql/reference/list-of-extensions. you may need to apply a maintenance upgrade to get it in your existing service.

**Comment by nikhilshinday** [comment]
anyone have an update on Aurora pgvector support on RDS?

**Comment by guillim** [comment]
> It seems to me that Google has recently secretly added support for `pgvector 0.4.2` in `POSTGRES_15_2.R20230530.01_00` maintenance update! The [list of extensions in the documentation](https://cloud.google.com/sql/docs/postgres/extensions#miscellaneous-extensions) has not been updated yet to reflect this but running `SELECT * FROM pg_available_extensions;` on a recently updated Cloud SQL instance shows this:
> 
> ![CleanShot 2023-06-19 at 16 56 34](https://user-images.githubusercontent.com/613647/246866075-d156e8f1-bc65-4abf-b395-2bdfa2d67f07.png)
> 
> Yas!

It’s now in the official documentation !!!

**Comment by developeralim** [comment]
![image](https://github.com/pgvector/pgvector/assets/95765755/bc36e41b-f059-400e-a71e-3d437e9b3d00)
I faced this issue when i was installing pgvector. i am using windows 10 . can anyone help me please

**Comment by pabloem** [comment]
Also Google's AlloyDB has added support for PGVector: https://cloud.google.com/blog/products/databases/announcing-vector-support-in-postgresql-services-to-power-ai-enabled-applications

**Comment by enchorb** [comment]
+1 pgvector for AWS Aurora

**Comment by disbelief** [comment]
So I guess this announcement from AWS on May 3 about pgvector support doesn't apply to Aurora? https://aws.amazon.com/about-aws/whats-new/2023/05/amazon-rds-postgresql-pgvector-ml-model-integration/

**Comment by enchorb** [comment]
Correct, pgvector works on Amazon RDS for PostgresSQL but not on Amazon Aurora PostgresSQL.

**Comment by retendo** [comment]
Although it is not documented, this seems to work on DigitalOcean Managed Postgres as well now. At least with Postgres 15. I didn't work with Postgres 12 though.

**Comment by ganesh-rao** [comment]
I believe AWS now supports pgvector on Aurora PostgresSQL:

https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Updates.html#AuroraPostgreSQL.Updates.20180305.153X

It's also listed here for Postgres 15.3:
https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Extensions.html#AuroraPostgreSQL.Extensions.15

**Comment by schoblaska** [comment]
DigitalOcean now lists pgvector on their page of supported extensions (for PostgreSQL 13 and newer): https://docs.digitalocean.com/products/databases/postgresql/details/supported-extensions/

**Comment by amadeus633** [comment]
> I believe AWS now supports pgvector on Aurora PostgresSQL:
> 
> https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Updates.html#AuroraPostgreSQL.Updates.20180305.153X
> 
> It's also listed here for Postgres 15.3: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Extensions.html#AuroraPostgreSQL.Extensions.15

Do you know if this includes Postgres on AWS Lightsail?

**Comment by StubbornDeer** [comment]
AWS Postgres 14: https://docs.aws.amazon.com/AmazonRDS/latest/PostgreSQLReleaseNotes/postgresql-extensions.html#postgresql-extensions-14x

**Comment by vishnu-narayanan** [comment]
AWS Postgres 13: https://docs.aws.amazon.com/AmazonRDS/latest/PostgreSQLReleaseNotes/postgresql-extensions.html#postgresql-extensions-13x

**Comment by shayons** [comment]
The pgvector extension is available on Aurora PostgreSQL as of July 13 (https://aws.amazon.com/about-aws/whats-new/2023/07/amazon-aurora-postgresql-pgvector-vector-storage-similarity-search) on versions 15.3, 14.8, 13.11, 12.15 and higher. Here's a blog that shows you how to build an AI powered chat bot application and perform sentiment analysis using pgvector and Amazon Aurora PostgreSQL - https://aws.amazon.com/blogs/database/leverage-pgvector-and-amazon-aurora-postgresql-for-natural-language-processing-chatbots-and-sentiment-analysis/

**Comment by ClearedFram3** [comment]
(off topic) @guillim is the GUI you show something from google cloud or a standalone product?

**Comment by jdowning** [comment]
Heroku Postges supports `pgvector`:

* https://github.com/heroku/roadmap/issues/156
* https://devcenter.heroku.com/changelog-items/2687

**Comment by jcrupi** [comment]
Looks like Azure Postgres Flex Server only supports pgvector .4,0

**Comment by gulcin** [comment]
EDB's [BigAnimal Postgres-as-a-Service](https://www.enterprisedb.com/products/biganimal-cloud-postgresql) supports pgvector:

- [https://www.enterprisedb.com/blog/what-is-pgvector](https://www.enterprisedb.com/blog/what-is-pgvector)

> BigAnimal is EDB’s managed Postgres service that allows you to run Postgres on all major cloud platforms. Now, you can enable pgvector on your databases and start experimenting!

**Comment by Kikiodazie** [comment]
Hey everyone, Severalnines' [ClusterControl](https://severalnines.com/clustercontrol/) now also supports `pgvector` in its PostgreSQL offering. 

See how here: https://docs.severalnines.com/docs/knowledgebase-tutorials/clustercontrol/tutorials/deploying-postgresql-cluster-with-pgvector-extension/.

**Comment by nicholasdly** [comment]
pgvector is supported on [Railway](https://railway.app/) via their official [pgvector project template](https://railway.app/template/3jJFCA), which has been updated to the latest version today!

(Thanks @brody192!)

**Comment by brody192** [comment]
thanks for the shout-out, but I feel it necessary to point out that it's not official, it's not published by Railway or pgvector themselves, I'm just a community member :)

**Comment by ikoif** [comment]
The [Heroku Postgres](https://devcenter.heroku.com/articles/pgvector-heroku-postgres) add-on seems to support it for tiers starting from **standard**.

**Comment by fijiwebdesign** [comment]
https://nexai.site now supports pgvector as a projects semantic database. 

<img width="1411" alt="nexai-dashboard-settings" src="https://github.com/pgvector/pgvector/assets/809011/973da253-070a-4444-a91e-20797f350a7f">

**Comment by philippemnoel** [comment]
@paradedb --> https://github.com/paradedb/paradedb supports `pgvector` as well.

**Comment by nclavaud** [comment]
[Scalingo](https://scalingo.com/) also announced support for pgvector:
- blog post: https://scalingo.com/blog/postgresql-15-available-with-pgvector
- documentation: https://doc.scalingo.com/databases/postgresql/pgvector

**Comment by PSYdex-dev** [comment]
```
[Zerops](https://zerops.io) also added support for pgvector (they recently posted about it on their discord server) 💯 

- Discord Server: https://discord.gg/xxzmJSDKPT

```

**Comment by StefanTerdell** [comment]
[Scaleway](https://www.scaleway.com/en/database/) (EU cloud provider) has had pgvector extension in their managed psql offering since november 2023. [Changelog entry](https://www.scaleway.com/en/docs/changelog/?product=postgresql-and-mysql&status=added#postgresql-pgvector-is-now-available)

It's been working great for me. I am otherwise unaffiliated.

**Comment by aravindputrevu** [comment]
Don't you folks think this should be a page in the repo and a issue?

**Comment by Buthrakaur** [comment]
https://aws.amazon.com/about-aws/whats-new/2023/05/amazon-rds-postgresql-pgvector-ml-model-integration/
> The pgvector extension is available on all database instances in Amazon RDS running PostgreSQL 15.2 and higher in all AWS Regions, including the AWS GovCloud (US) Regions.

**Comment by gideongrinberg** [comment]
DigitalOcean supports pgvector: https://docs.digitalocean.com/products/databases/postgresql/details/supported-extensions/

**Comment by ankane** [comment]
Thanks all, this has been one of my favorite threads. Now that there's support on many hosted providers, closing this out.
